File: Samples/Android/app/src/main/java/com/tencent/voiceai/demo/ui/components/PageScaffold.kt
// Copyright (c) 2026 Tencent. All rights reserved.
// VoiceAI Demo - 通用页面脚手架（顶部栏 + 返回）

package com.tencent.voiceai.demo.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.RowScope
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.tencent.voiceai.demo.ui.theme.BackgroundColor
import com.tencent.voiceai.demo.ui.theme.TextPrimary
import com.tencent.voiceai.demo.ui.theme.TextSecondary

/**
 * 二级页面通用脚手架：亮色背景 + 顶部标题栏（含返回按钮）+ 内容区。
 *
 * @param actions 标题栏右侧的操作按钮（RowScope），例如电话、朗读开关。
 */
@Composable
fun PageScaffold(
    title: String,
    onBack: () -> Unit,
    actions: @Composable (RowScope.() -> Unit)? = null,
    content: @Composable () -> Unit
) {
    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(BackgroundColor)
    ) {
        TopBar(title = title, onBack = onBack, actions = actions)
        Box(modifier = Modifier.fillMaxSize()) {
            content()
        }
    }
}

@Composable
private fun TopBar(
    title: String,
    onBack: () -> Unit,
    actions: @Composable (RowScope.() -> Unit)? = null
) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .padding(horizontal = 16.dp, vertical = 14.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.Start
    ) {
        Box(
            modifier = Modifier
                .size(40.dp)
                .clip(CircleShape)
                .background(Color(0xFFE7ECF7))
                .clickable(onClick = onBack),
            contentAlignment = Alignment.Center
        ) {
            Icon(
                imageVector = Icons.AutoMirrored.Filled.ArrowBack,
                contentDescription = "返回",
                tint = Color(0xFF4B5468),
                modifier = Modifier.size(22.dp)
            )
        }

        Spacer(modifier = Modifier.width(14.dp))

        Text(
            text = title,
            fontSize = 20.sp,
            fontWeight = FontWeight.Bold,
            color = TextPrimary
        )

        Spacer(modifier = Modifier.weight(1f))

        if (actions != null) {
            actions()
        }
    }
}

/**
 * 占位内容：功能开发中提示。
 */
@Composable
fun PlaceholderContent(description: String) {
    Column(
        modifier = Modifier
            .fillMaxSize()
            .padding(32.dp),
        verticalArrangement = Arrangement.Center,
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        Text(
            text = "功能开发中",
            fontSize = 18.sp,
            fontWeight = FontWeight.Bold,
            color = TextPrimary
        )
        Spacer(modifier = Modifier.height(10.dp))
        Text(
            text = description,
            fontSize = 13.sp,
            color = TextSecondary
        )
    }
}

File: Samples/iOS/VoiceAIKitDemo/Demo/Components/PageScaffold.swift
// Copyright (c) 2026 Tencent. All rights reserved.
// VoiceAI Demo - 通用页面脚手架 (对齐 Android ui/components/PageScaffold.kt)

import SwiftUI

/// 带标题栏的通用页面容器：左侧返回按钮 + 居中标题 + 右侧可选操作区。
struct PageScaffold<TrailingContent: View, Body: View>: View {
    let title: String
    let onBack: () -> Void
    @ViewBuilder let trailing: () -> TrailingContent
    @ViewBuilder let content: () -> Body

    init(title: String,
         onBack: @escaping () -> Void,
         @ViewBuilder trailing: @escaping () -> TrailingContent = { EmptyView() },
         @ViewBuilder content: @escaping () -> Body) {
        self.title = title
        self.onBack = onBack
        self.trailing = trailing
        self.content = content
    }

    var body: some View {
        VStack(spacing: 0) {
            ZStack {
                Text(title)
                    .font(.system(size: 18, weight: .semibold))
                    .foregroundColor(Theme.textPrimary)
                HStack {
                    Button(action: onBack) {
                        Image(systemName: "chevron.left")
                            .font(.system(size: 20, weight: .semibold))
                            .foregroundColor(Theme.textPrimary)
                            .frame(width: 40, height: 40)
                    }
                    Spacer()
                    trailing()
                }
            }
            .padding(.horizontal, 8)
            .frame(height: 52)
            .background(Theme.background)

            content()
                .frame(maxWidth: .infinity, maxHeight: .infinity)
        }
        .background(Theme.background)
    }
}
